server/src/main/java/org/apache/druid/server/coordinator/loading/LoadQueueTaskMaster.java

```diff
@@ -81,6 +91,50 @@ public LoadQueuePeon giveMePeon(ImmutableDruidServer server)
     }
   }
 
+  public Map<String, LoadQueuePeon> getAllPeons()
+  {
+    return loadManagementPeons;
+  }
+
+  public LoadQueuePeon getPeonForServer(ImmutableDruidServer server)
+  {
+    return loadManagementPeons.get(server.getName());
+  }
+
+  public void startPeonsForNewServers(List<ImmutableDruidServer> currentServers)
+  {
+    for (ImmutableDruidServer server : currentServers) {
+      loadManagementPeons.computeIfAbsent(server.getName(), serverName -> {
+        LoadQueuePeon loadQueuePeon = createPeon(server);
+        loadQueuePeon.start();
+        log.debug("Created LoadQueuePeon for server[%s].", server.getName());
+        return loadQueuePeon;
+      });
+    }
+  }
+
+  public void stopPeonsForDisappearedServers(List<ImmutableDruidServer> servers)
+  {
+    final Set<String> disappearedServers = Sets.newHashSet(loadManagementPeons.keySet());
```

Comment:
From the method name I assumed that we would stopping peons for the sets of servers passed, instead it seems to be doing the opposite. Can you please add a comment?

Reply:
Yeah, fair point. I didn't like the structure of this method either.
Please let me know if there are any further comments. I will address all of them together.